torchaudio/datasets/cmudict.py

```diff
@@ -0,0 +1,92 @@
+import re
+from pathlib import Path
+from typing import Tuple, Union, List, Dict
+
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import download_url
+
+_CHECKSUMS = {
+    "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b":
+    "825f4ebd9183f2417df9f067a9cabe86",
+    "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols":
+    "385e490aabc71b48e772118e3d02923e",
+}
+
+
+class CMUDICT(Dataset):
+    """Create a Dataset for CMU Pronouncing Dictionary (CMUDict).
+
+    Args:
+        root (str or Path): Path to the directory where the dataset is found or downloaded.
+        url (str, optional):
+            The URL to download the dictionary from.
+            (default: ``"http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b"``)
+        url_symbols (str, optional):
+            The URL to download the list of symbols from.
+            (default: ``"http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols"``)
+        download (bool, optional):
+            Whether to download the dataset if it is not found at root path. (default: ``False``).
+    """
+
+    def __init__(self,
+                 root: Union[str, Path],
+                 url: str = "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b",
+                 url_symbols: str = "http://svn.code.sf.net/p/cmusphinx/code/trunk/cmudict/cmudict-0.7b.symbols",
+                 download: bool = False) -> None:
+
+        # Get string representation of 'root' in case Path object is passed
+        root = os.fspath(root)
+        if not os.path.isdir(root):
+            os.mkdir(root)
+
+        if download:
+            if not os.path.isdir(root):
+                checksum = _CHECKSUMS.get(url, None)
+                download_url(url, root, hash_value=checksum, hash_type="md5")
+                checksum = _CHECKSUMS.get(url_symbols, None)
+                download_url(url_symbols, root, hash_value=checksum, hash_type="md5")
+
+        self._root_path = root
+        basename = os.path.basename(url)
+        basename_symbols = os.path.basename(url_symbols)
+
+        with open(os.path.join(self._root_path, basename_symbols), "r") as text:
+            self._symbols = text.readlines()
+
+        with open(os.path.join(self._root_path, basename), "r") as text:
+            self._dictionary = self._parse_dictionary(text.readlines())
+
+        self._available_words = list(self._dictionary.keys())
+
+    def _parse_dictionary(self, lines: List[str]):
+        _alt_re = re.compile(r'\([0-9]+\)')
+        cmudict: List[Tuple[str, List[str]]] = list
+        for line in lines:
+            if len(line) and line[:3] != ";;;":  # ignore comments
+                parts = line.strip().split('  ')
+                # if a word have multiple pronunciations, there will be (number) appended to it
+                # for example, DATAPOINTS(1)
+                word = re.sub(_alt_re, '', parts[0])
```

Comment:
If I am reading this regex correctly, this removes the number.
I guess the comment, "there will be (number) appended" is referring to the original form, and I found it confusing. Can you, instead, explain what is done (and possibly why)?

Reply:
A more detailed explanation is [here](https://github.com/pytorch/audio/pull/1627/files#diff-50ee900e0897ac7528dccd9bdd36d5f0c7cb2d45310aefb46a3bf281f73d88c2R78). Please let me know if it is clear.
Thanks.